src/main/java/com/celements/payment/service/ComputopService.java

```diff
@@ -92,8 +91,126 @@ String hashPaymentData(@NotNull String paymentData) {
+        currency);
+    Map<String, String> encryptedData = new HashMap<>();
+    encryptedData.put(FORM_INPUT_NAME_LENGTH, Integer.toString(dataPlainText.length()));
+    String encyrptedData = encryptString(dataPlainText.getBytes(), getBlowfishKey());
+    if (encyrptedData != null) {
+      encryptedData.put(FORM_INPUT_NAME_DATA, encyrptedData);
+    }
+    return encryptedData;
+  }
+
+  @Override
+  public Map<String, String> decryptCallbackData(String encryptedCallback, int plainDataLength) {
+    checkNotNull(encryptedCallback);
+    byte[] decryptedData = decryptString(encryptedCallback, plainDataLength, getBlowfishKey());
+    Map<String, String> callbackData = new HashMap<>();
+    if (decryptedData != null) {
+      for (String parameter : (new String(decryptedData)).split("&")) {
+        String[] keyValue = parameter.split("=", 2);
+        String key = keyValue[0].toLowerCase();
+        String value = "";
+        if (keyValue.length > 1) {
+          value = keyValue[1];
+        }
+        callbackData.put(key, value);
+      }
+    }
+    return callbackData;
+  }
+
+  String getPaymentDataPlainString(String transactionId, String orderDescription, BigDecimal amount,
+      String currency) {
+    checkNotNull(transactionId);
+    checkNotNull(amount);
+    currency = Optional.fromNullable(currency).or(DEFAULT_CURRENCY);
+    String merchantId = getMerchantId();
+    StringBuilder sb = new StringBuilder();
+    sb.append(FORM_INPUT_NAME_MERCHANT_ID).append("=").append(merchantId);
+    sb.append("&").append(FORM_INPUT_NAME_TRANS_ID).append("=").append(transactionId);
+    sb.append("&").append(FORM_INPUT_NAME_AMOUNT).append("=").append(getAmount(amount));
+    sb.append("&").append(FORM_INPUT_NAME_CURRENCY).append("=").append(currency);
+    sb.append("&").append(FORM_INPUT_NAME_DESCRIPTION).append("=").append(orderDescription);
+    sb.append("&").append(FORM_INPUT_NAME_HMAC);
+    sb.append("=").append(getPaymentDataHmac(null, transactionId, merchantId, amount, currency));
+    sb.append("&").append(ReturnUrl.SUCCESS.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.SUCCESS));
+    sb.append("&").append(ReturnUrl.FAILURE.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.FAILURE));
+    sb.append("&").append(ReturnUrl.CALLBACK.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.CALLBACK));
+    return sb.toString();
+  }
+
+  String encryptString(byte[] plainText, final SecretKey key) {
+    LOGGER.debug("encrypting plain [{}]", plainText);
+    try {
+      String cipherText = BaseEncoding.base16().encode(getCipher(Cipher.ENCRYPT_MODE,
+          BLOWFISH_ECB_PADDED, key).doFinal(plainText));
+      LOGGER.debug("encrypted cipher [{}]", cipherText);
+      return cipherText;
+    } catch (IllegalBlockSizeException | BadPaddingException | NoSuchPaddingException excp) {
+      LOGGER.error("Problem while encrypting payment data", excp);
```

Comment:
I would prefer `encryptPaymentData`/`encryptString` to fail with an exception e.g. `ComputopEncryptionException` if the encryption fails